tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/client/SlowConsumerTest.java

```diff
@@ -612,6 +612,45 @@ public void testMinuteSurviving() throws Exception {
       Wait.assertEquals(0, queue::getConsumerCount);
    }
 
+
+   @Test
+   public void testKilledOnNoMessagesSoCanBeRebalanced() throws Exception {
+      AddressSettings addressSettings = new AddressSettings();
+      addressSettings.setSlowConsumerCheckPeriod(2);
+      addressSettings.setSlowConsumerThresholdMeasurementUnit(MESSAGES_PER_SECOND);
+      addressSettings.setSlowConsumerThreshold(0); // if there are no messages pending, kill me
+      addressSettings.setSlowConsumerPolicy(SlowConsumerPolicy.KILL);
+
+      server.getAddressSettingsRepository().removeMatch(QUEUE.toString());
+      server.getAddressSettingsRepository().addMatch(QUEUE.toString(), addressSettings);
+
+
+      ClientSessionFactory sf = createSessionFactory(locator);
+
+      ClientSession session = addClientSession(sf.createSession(false, true, true, false));
+
+      ClientProducer producer = addClientProducer(session.createProducer(QUEUE));
+
+      int messages = 1;
+
+      for (int i = 0; i < messages; i++) {
+         producer.send(session.createMessage(true));
+      }
+      session.commit();
+
+      ConcurrentHashSet<ClientMessage> receivedMessages = new ConcurrentHashSet<>();
+      FixedRateConsumer consumer = new FixedRateConsumer(10, MESSAGES_PER_SECOND, receivedMessages, sf, QUEUE, 0);
+      consumer.start();
+
+      Queue queue = server.locateQueue(QUEUE);
+      Wait.assertEquals(1, queue::getConsumerCount);
```

Comment:
This could be time sensitive, what about replacing FixedRateConsumer with a manual consumer and check consumerCount before receiving the message?

Reply:
is it the locate queue that may not yet be present?

Reply:
I suppose there is no need for a fixed rate consumer tho, it is irrelevant to the test. a simple consumer would suffice

Reply:
I have tidied it up a bit, not 100% I captured your feedback!